firefox_ui_harness/runners/update.py

```diff
@@ -42,7 +42,7 @@ def __init__(self, **kwargs):
 
     def duplicate_application(self, application_folder):
         """Creates a copy of the specified binary."""
-        target_folder = os.path.join(self.workspace, 'binary.backup')
+        target_folder = os.path.join(self.workspace_path, 'binary.backup')
```

Comment:
My understanding of your comments is that firefox-ui-test-specific installer/binary stuff should go into TMP if no workspace is set.